Document autodevice feature (#440)
AdguardTeam/KnowledgeBaseDNS: docs/dns-client/configuration.md

Reviewer: If `autodevice.enabled` set to `true` for this group, the address should be a URL with one of `https`, `tls`, or `quic` scheme.

- `groups`: The set of upstream servers keyed by the group’s name. It has the following properties:

    - `address`: The upstream server’s address. If `autodevice.enabled` set to `true` for this group, the address should be a URL with one of `https`, `tls`, or `quic` scheme.

      **Example:** `'8.8.8.8:53'`

>     - `autodevice`: Represents an automatic connection of a device.

Reviewer: At last, I'd add a reference to the `private-dns/connect-devices/other-options/automatic-connection`.

      :::note

      The autodevice option must be used only for AdGuard DNS upstreams. Otherwise, we can’t guarantee proper work.

      :::

      It has the following properties:

        - `enabled`: Defines whether all clients within the current group can be connected automatically.

>         :::info

Reviewer 2: Wrong indentation level breaking the list.

- `groups`: The set of upstream servers keyed by the group’s name. It has the following properties:

    - `address`: The upstream server’s address. If `autodevice.enabled` set to `true` for this group, the address should be a URL with one of `https`, `tls`, or `quic` scheme.

      **Example:** `'8.8.8.8:53'`

    - `autodevice`: Represents an automatic connection of a device.

      :::note

      The autodevice option must be used only for AdGuard DNS upstreams. Otherwise, we can’t guarantee proper work.

>       :::

Reviewer: What is that?

Author: I am not following you. Explain.

      It has the following properties:

>         :::info

Reviewer: Put it under the `enabled` list item.

        The predefined `private` group must have `enabled` set to false, as it doesn't support autodevice yet.

        :::

        - `enabled`: Defines whether all clients within the current group can be connected automatically.

        - `profile_id`: [ID of a profile][profile-id], in which new devices will be added.

        - `device_type`: A [type of device][device-type] which will be created for new clients.

>         [profile-id]:  /private-dns/solving-problems/automatic-devices/#dns-server-id

Reviewer: Put it to the end of the second-level (`##`) section.  See https://github.com/AdguardTeam/CodeGuidelines/blob/master/Go/Markdown.md#li-7b291978.

      **Example:** `'8.8.8.8:53'`

>       :::note

Reviewer: I think this admonition should be within the `autodevice` item.

Author: Done.

>       The autodevice option must be used only for AdGuard DNS upstreams. Otherwise, we can't guarantee proper work.

Reviewer 3: ```suggestion
      The autodevice option must be used only for AdGuard DNS upstreams. Otherwise, we can’t guarantee proper work.
```

      :::

    - `autodevice`: Represents an automatic connection of a device. It has the following properties:

        - `enabled`: Defines whether all clients within the current group can be connected automatically.

        - `profile_id`: ID of a profile, in which new devices will be added.

        - `device_type`: A type of device which will be created for new clients.

        **Property example:**

        ```yaml
          'autodevice':
              - enabled: true
              - profile_id: 'defa5678'

>               - device_type: '1'

Reviewer 2: 1. Unnecessary top-level indentation.
2. Invalid `device_type`.

      **Example:** `'8.8.8.8:53'`

    - `autodevice`: Represents an automatic connection of a device.

      :::note

      The autodevice option must be used only for AdGuard DNS upstreams. Otherwise, we can’t guarantee proper work.

      :::

      It has the following properties:

>         - `enabled`: Defines whether all clients within the current group can be connected automatically.

Reviewer: Add an `:::info` admonition, that the predefined `private` group must have it set to `false`, as it doesn't support autodevice yet.

>         - `profile_id`: ID of a profile, in which new devices will be added.

Reviewer: It would actually be helpful to reference the possible values of `device_type` and the description of `profile_id` from the AGDNS documentation.  I've definitely saw them somewhere.

      It has the following properties:

        :::info

>         The predefined private group must have `enabled` set to false, as it doesn't support autodevice yet.

Reviewer: `private` is the literal name, please format it at least monospaced.

        :::

        - `enabled`: Defines whether all clients within the current group can be connected automatically.

        - `profile_id`: [ID of a profile][profile-id], in which new devices will be added.

        - `device_type`: A [type of device][device-type] which will be created for new clients.

>         [profile-id]:  [https://adguard-dns.io/kb/private-dns/solving-problems/automatic-devices/#dns-server-id]

Reviewer: 1. This is not how references are written.
1. These pages exist within the same project.  Please investigate how other references to the pages within the knowledge base are made.

  :::info

>   `groups` should contain at least a single entry named `default`, and optionally a single entry named `private`, both should have no `match` property. The `private` group is also used in defining of the HumanID for clients, if it is not defined, an alternative way of generation is used: the HumanID is formed from the IP address.

Reviewer 2: This one new sentence seems like it really wants to be two sentences.  `for clients. If it is`.

>   `groups` should contain at least a single entry named `default`, and optionally a single entry named `private`, both should have no `match` property. The `private` group is also used in defining of the HumanID for clients. If it is not defined, an alternative way of generation is used: the HumanID is formed from the IP address.

Reviewer 3: ```suggestion
  `groups` should contain at least a single entry named `default`, and optionally a single entry named `private`, both should have no `match` property. The `private` group is also used to define the HumanID for clients. If it is not defined, an alternative generation method is used, whereby the HumanID is formed from the IP address.
```

  :::info

>   `groups` should contain at least a single entry named `default`, and optionally a single entry named `private`, both should have no `match` property. The `private` group is also used to define the HumanID for clients. If it is not defined, an alternative generation method is used, whereby the HumanID is formed from the IP address.

Reviewer: > The `private` group is also used to define the HumanID for clients **created by `autodevice` feature**.
